src/wp-includes/blocks/index.php

```diff
@@ -12,6 +12,77 @@
 require BLOCKS_PATH . 'widget-group.php';
 require BLOCKS_PATH . 'require-dynamic-blocks.php';
 
+/**
+ * Registers core block style handles.
+ *
+ * While {@see register_block_style_handle()} is typically used for that, the way it is
+ * implemented is inefficient for core block styles. Registering those style handles here
+ * avoids unnecessary logic and filesystem lookups in the other function.
+ *
+ * @since 6.3.0
+ */
+function register_core_block_style_handles() {
+	if ( ! wp_should_load_separate_core_block_assets() ) {
+		return;
+	}
+
+	static $core_blocks_meta;
+	if ( ! $core_blocks_meta ) {
+		$core_blocks_meta = require ABSPATH . WPINC . '/blocks/blocks-json.php';
+	}
+
+	$includes_url   = includes_url();
+	$suffix         = wp_scripts_get_suffix();
+	$wp_styles      = wp_styles();
+	$style_fields   = array(
+		'style',
+		'editorStyle',
+	);
+	$transient_name = 'block_styles';
+	$files          = get_transient( $transient_name );
+	if ( ! $files ) {
+		$files = glob( __DIR__ . "/**/style*{$suffix}.css" );
+		set_transient( $transient_name, $files );
+	}
+
+	foreach ( $core_blocks_meta as $name => $schema ) {
+		if ( ! isset( $schema['style'] ) ) {
+			$schema['style'] = "wp-block-$name";
+		}
+		if ( ! isset( $schema['editorStyle'] ) ) {
+			$schema['editorStyle'] = "wp-block-{$name}-editor";
+		}
```

Comment:
Before this backfilling, we also need to run the `block_type_metadata` filter, otherwise the data may become inconsistent.
```suggestion
		/** This filter is documented in wp-includes/blocks.php */
		$schema = apply_filters( 'block_type_metadata', $schema );

		// Backfill these properties similar to `register_block_type_from_metadata()`.
		if ( ! isset( $schema['style'] ) ) {
			$schema['style'] = "wp-block-$name";
		}
		if ( ! isset( $schema['editorStyle'] ) ) {
			$schema['editorStyle'] = "wp-block-{$name}-editor";
		}
```